src/main/kotlin/com/dansplugins/factionsystem/locks/MfLockService.kt

```diff
@@ -43,6 +53,39 @@ class MfLockService(private val plugin: MedievalFactions, private val repository
         ServiceFailure(exception.toServiceFailureType(), "Service error: ${exception.message}", exception)
     }
 
+    fun unlock(block: Block): MfUnlockResult {
+        val blockData = block.blockData
+        val holder = (block.state as? Chest)?.inventory?.holder
+        val blocks = if (blockData is Bisected) {
+            if (blockData.half == BOTTOM) {
+                listOf(block, block.getRelative(UP))
+            } else {
+                listOf(block, block.getRelative(DOWN))
+            }
+        } else if (holder is DoubleChest) {
+            val left = holder.leftSide as? Chest
+            val right = holder.rightSide as? Chest
+            listOfNotNull(left?.block, right?.block)
+        } else {
+            listOf(block)
+        }
+        var toReturn = MfUnlockResult.SUCCESS
+        plugin.server.scheduler.runTaskAsynchronously(plugin, Runnable {
+            val lockedBlocks = blocks.mapNotNull { getLockedBlock(MfBlockPosition.fromBukkitBlock(it)) }
+            val lockedBlock = lockedBlocks.firstOrNull()
+            if (lockedBlock == null) {
+                toReturn = MfUnlockResult.NOT_LOCKED
+                return@Runnable
+            }
+            delete(lockedBlock.block).onFailure {
+                plugin.logger.log(SEVERE, "Failed to delete block: ${it.reason.message}", it.reason.cause)
+                toReturn = MfUnlockResult.FAILURE
+                return@Runnable
+            }
+        })
+        return toReturn
```

Comment:
This will basically always return SUCCESS because the asynchronous task will fire after returning.
Perhaps pass in a callback that receives the result?